### DISCUSSION TITLE
Need to extent existing EndPoint Activity

### DISCUSSION THREAD

**Discussion by Lalitaru143** [elsa-workflows/elsa-core#1212] [Q&A]
Working on the custom activity and want to have similar activity like HttpEndPoint. So that directly call it as an API.

Can you please provide me any solution. ?

**Comment by sfmskywalker** [comment]
Depending on your exact use case, you could do something like the Webhooks module is doing. Instead of inheriting from `HttpEndpoint`, it uses an `IActivityTypeProvider` implementation that dynamically yields new activity types that reuse `HttpEndpoint`.

If you do want to derive a new activity type from `HttpEndpoint` directly, you could probably do it. Just make sure that when you implement your bookmark provider that it is associated with the `HttpEndpoint` activity - the HTTP middleware relies on that.

Alternatively, you might go a different route altogether and simply implement an API endpoint (as an ASP.NET Core controller, middleware or route endpoint) that triggers workflows based on your custom activity. I've written some documentation about that process here: https://elsa-workflows.github.io/elsa-core/docs/next/guides/guides-blocking-activities

**Reply by sfmskywalker** [reply]
See also: https://stackoverflow.com/questions/68153272/with-elsa-workflow-2-0-how-to-create-an-activity-based-on-httpendpoint
